File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/sqlite/EnumType.scala
package dev.typr.foundationssc.docs.sqlite
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object EnumType:
  // start
  // Plain Scala 3 enum
  enum Status:
    case PENDING, ACTIVE, COMPLETED

  // SQLite has no native enum — pair with a CHECK (col IN (...)) constraint in DDL.
  val statusType: SqliteType[Status] = SqliteTypes.ofEnum(Status.values)
  // stop

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/sqlite/EnumType.kt
package dev.typr.foundationskt.docs.sqlite

import dev.typr.foundationskt.*

@Suppress("unused")
class EnumType {
    //start
    enum class Status { PENDING, ACTIVE, COMPLETED }

    // SQLite has no native enum — pair with `CHECK (col IN ('PENDING','ACTIVE','COMPLETED'))` in DDL.
    val statusType: SqliteType<Status> = SqliteTypes.ofEnum<Status>()
    //stop
}
